components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -409,37 +409,62 @@ bool AdsServiceImpl::StartService() {
   bat_ads_service_.set_connection_error_handler(
       base::Bind(&AdsServiceImpl::MaybeStart, AsWeakPtr(), true));
 
-  bool is_production = false;
+  UpdateIsProductionFlag();
+  UpdateIsDebugFlag();
+  UpdateIsTestingFlag();
+
+  return true;
+}
+
+#if defined(OS_ANDROID)
+void AdsServiceImpl::UpdateIsProductionFlag() {
```

Comment:
Seems like this could all be done with just the single function, unless there are other reasons for splitting it out like this.

Reply:
I did that originally and due to #if etc. the code was harder to read

Reply:
OK, gotcha.

Reply:
Eventually this code should be abstracted out of the ads service, this changed was mainly to add support for Android Ads